### PR TITLE
matplotlib based coloring viewer

### PR DESCRIPTION
**Pull request by naylor-b**
- also backed out some recent changes to the randomization we use to generate the sparsity matrix.

### CODE REVIEW

File: openmdao/jacobians/jacobian.py [hunk 1]
```diff
@@ -352,6 +349,8 @@ def _compute_sparsity(self, ordered_of_info, ordered_wrt_info, num_full_jacs, to
                     else:
                         J[roffset:rend, coffset:cend] = summ[key]
 
+        J /= num_full_jacs
+
```
Comment (Kenneth-T-Moore) [inline openmdao/jacobians/jacobian.py]: I believe J *= (1.0/num_full_jacs) is a lot faster.

File: openmdao/docs/features/core_features/working_with_derivatives/simul_derivs.rst [hunk 2]
```diff
@@ -176,6 +176,19 @@ option as follows:
...
+See the figure :ref:`here <fig-coloring-view>`.
+
+
+A text-based view is also available using the `--jtext` arg.  For example:
+
+
+.. code-block:: none
+
+    openmdao total_coloring <your_script_name> --jtext
+
```
Comment (Kenneth-T-Moore) [inline openmdao/docs/features/core_features/working_with_derivatives/simul_derivs.rst]: Anyway to also make this work from inside your run script, much like `view_model(prob)`?

File: openmdao/utils/coloring.py [hunk 3]
```diff
@@ -1173,6 +1334,8 @@ def _get_bool_total_jac(prob, num_full_jacs=_DEF_COMP_SPARSITY_ARGS['num_full_ja
                 fullJ += np.abs(J)
         elapsed = time.time() - start_time
 
+    fullJ /= num_full_jacs
```
Comment (Kenneth-T-Moore) [inline openmdao/utils/coloring.py]: Also here.

File: openmdao/docs/features/core_features/working_with_derivatives/simul_derivs.rst (hunk 2 above)
Comment (naylor-b) [reply openmdao/docs/features/core_features/working_with_derivatives/simul_derivs.rst]: Programmatically you can call display() on the Coloring object.  The coloring object is accessible via system._coloring_info['coloring'] or can be loaded from a file using Coloring.load(fname)

File: openmdao/jacobians/jacobian.py (hunk 1 above)
Comment (naylor-b) [reply openmdao/jacobians/jacobian.py]: done

File: openmdao/utils/coloring.py (hunk 3 above)
Comment (naylor-b) [reply openmdao/utils/coloring.py]: done

**Pull request merged**